src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -549,7 +549,10 @@ def _forward(self, model_inputs, return_timestamps=False, generate_kwargs=None):
                     generate_kwargs["return_token_timestamps"] = True
 
                     if stride is not None:
-                        generate_kwargs["num_frames"] = stride[0] // self.feature_extractor.hop_length
+                        if isinstance(stride, tuple):
+                            generate_kwargs["num_frames"] = stride[0] // self.feature_extractor.hop_length
+                        else:
+                            generate_kwargs["num_frames"] = stride[0][0] // self.feature_extractor.hop_length
```

Comment:
Just as a double check: is this the intended behaviour for batched inference @xenova? i.e. we take the first possible stride value in the batch? Or do we actually need to pass a **tuple** of `num_frames` to the generate method? i.e. one value of `num_frames` for each item in the batch?

Reply:
Passing a tuple of num frames would be problematic, since we would end up with a ragged tensor when we slice the attention weights (slice a tensor by different lengths): https://github.com/huggingface/transformers/blob/a5e6df82c00cb53ffe863008cbbedd813bcc508b/src/transformers/models/whisper/modeling_whisper.py#L1836-L1837

=> should we set `num_frames=None` when we have a batch here of strides?

Reply:
> Passing a tuple of num frames would be problematic, since we would end up with a ragged tensor when we slice the attention weights (slice a tensor by different lengths)

Good spot 👀 The only case where this _should_ happen for audio segments that are not exactly a multiple of 30 seconds (since the last chunk will have a different number of frames).

In such a case, the only logical way to handle it would be to do it separately (i.e., the last chunk would have `batch_size=1`)... a nastly little edge case. WDYT?